beetsplug/mbsubmit.py

```diff
@@ -79,6 +407,36 @@ def print_tracks(self, session, task):
         for i in sorted(task.items, key=lambda i: i.track):
             print_data(None, i, self.config["format"].as_str())
 
+    def create_release_on_musicbrainz(self, session, task):
+        if not self._start_server():
+            return
+        task_key = str(uuid.uuid4())
+        token = jwt.encode(
+            {"task_key": task_key}, self.jwt_key, algorithm=self.jwt_algorithm
+        )
+
+        self.server_tasks[task_key] = build_formdata(
+            task=task,
+            redirect_uri=f"http://{self.server_hostname}:{self._server.server_port}/complete_add?token={token}",
+        )
+
+        url = f"http://{self.server_hostname}:{self._server.server_port}/add?token={token}"
+        if self.server_open_method == "open_browser":
+            webbrowser.open(url)
+        elif self.server_open_method == "show_link":
+            print_(f"Open the following URL in your browser: {url}")
+        else:
+            return
+
+        print_("Waiting for MusicBrainz release ID...")
+
+        while (mbid := self.server_task_results.get(task_key, None)) is None:
+            time.sleep(0.1)
+            # TODO need a better way to do this
```

Comment:
Not sure what to do here!

I want to wait until the release_mbid is available (by saving the release on musicbrainz). This works. What also works is creating a threading.Event(), triggering that in the RequestHandler, and waiting here.

BUT: In both cases the user can't exit with CTRL+c, which is bad...

Don't see a way to catch KeyboardInterrupt here either, since that's handled way further up... Can I somehow check if the current thread is aborted and should exit?

Reply:
Solved.